Sources/NetworkProtection/Diagnostics/NetworkProtectionConnectionTester.swift

```diff
@@ -28,6 +28,7 @@ import Common
 /// the HTTPs port (443) both with and without using the tunnel.  The tunnel connection will be considered to be disconnected
 /// whenever the regular connection works fine but the tunnel connection doesn't.
 ///
+@MainActor
```

Comment:
Considering the connection tester is run at very specific times when important things need to happen, we may as well mark it all `@MainActor`.